devito/interfaces.py

```diff
@@ -538,35 +562,39 @@ def dim(self):
     def forward(self):
         """Symbol for the time-forward state of the function"""
         i = int(self.time_order / 2) if self.time_order >= 2 else 1
+        _t = self.indices[0]
 
-        return self.subs(t, t + i * s)
+        return self.subs(_t, _t + i * s)
 
     @property
     def backward(self):
         """Symbol for the time-backward state of the function"""
         i = int(self.time_order / 2) if self.time_order >= 2 else 1
+        _t = self.indices[0]
 
-        return self.subs(t, t - i * s)
+        return self.subs(_t, _t - i * s)
 
     @property
     def dt(self):
         """Symbol for the first derivative wrt the time dimension"""
+        _t = self.indices[0]
         if self.time_order == 1:
             # This hack is needed for the first-order diffusion test
-            indices = [t, t + s]
+            indices = [_t, _t + s]
```

Comment:
Yes, but this change is only concerned with allowing different time symbols here. Actually changing the timestepping scheme here requires a little more thought and a lot more testing. Now that we have the `TimeAxis` object though we should aim to implement the classic Forward/Backward/Crank Nicholson timesteppers, as documented [here](https://en.wikipedia.org/wiki/Finite_difference_method).